### ISSUE TITLE
Preprocessor chain 'climate_statistics' and 'multimodel_statistics' fails as time dimension is not present

### ISSUE THREAD

**Issue by tomaslovato** [ESMValGroup/ESMValCore#1018]
The construction of a preprocessor that computes first `climate_statistics` and then `multimodel_statistics` fails with a missing `time` coordinate error from the latter.

This is reasonable as for individual model `climate_statistics` the time axis is replaced with the one corresponding to requested operation (either 'day_of_year', 'month_number', 'season_number'), but `multimodel_statistics` expects to receive a cube with the `time` coordinate

I made a test and it should be a big issue to replace the deletion of `time` coordinate from the cube after aggregation
https://github.com/ESMValGroup/ESMValCore/blob/7f8406bdafe898c05d37f398457023b228e79324/esmvalcore/preprocessor/_time.py#L537-L539

with a demotion of the cube `time` coordinate to auxiliary (time will have values coherent with the aggregation operation), as in the following 
```python
@@ -535,7 +535,8 @@ def climate_statistics(cube,
     clim_coord = _get_period_coord(cube, period, seasons)
     operator = get_iris_analysis_operation(operator)
     clim_cube = cube.aggregated_by(clim_coord, operator)
-    clim_cube.remove_coord('time')
+    iris.util.demote_dim_coord_to_aux_coord(clim_cube,'time')
     if clim_cube.coord(clim_coord.name()).is_monotonic():
         iris.util.promote_aux_coord_to_dim_coord(clim_cube, clim_coord.name())
     else:
```

**Comment by valeriupredoi** [comment]
cheers @tomaslovato - related to https://github.com/ESMValGroup/ESMValCore/issues/890 - I'll have to find some time next week sit down with the code and get all this jazz in (note the associated issues related to `multimodel`, there are a few)

**Comment by tomaslovato** [comment]
@valeriupredoi I had a look to changes of time handling in PR #1023, but yet the code for multimodel statistics requires that `time` is a coordinate of the cube
https://github.com/ESMValGroup/ESMValCore/blob/0c798617b885777a5ffa1697c39bb46974a73bab/esmvalcore/preprocessor/_multimodel.py#L208-L212

while `time` coordinate is removed when computing the `climate_statistics`
https://github.com/ESMValGroup/ESMValCore/blob/0c798617b885777a5ffa1697c39bb46974a73bab/esmvalcore/preprocessor/_time.py#L544-L546

I think that the solution proposed above in this discussion should be a viable fix.

**Comment by valeriupredoi** [comment]
looking at #968 I reckon we can do stuffs after merge @tomaslovato - but I like your enthusiasm and optimism :grin:

**Comment by tomaslovato** [comment]
As discussed in https://github.com/ESMValGroup/ESMValCore/pull/1095#issuecomment-832043564 and #1069, I'll open a PR to fix the time issue of `climate_statistics` as implemented in 3dad5db80c2
